runnc-cont/runnc_cont.go

```diff
@@ -72,40 +67,108 @@ type RunncCont struct {
 
 // NewRunncCont returns a brand new runnc-cont
 func NewRunncCont(cfg Config) (*RunncCont, error) {
-	if len(cfg.IPAddress) == 0 {
-		cfg.IPAddress = "10.0.0.2"
-	}
-	if cfg.IPMask == 0 {
-		cfg.IPMask = 24
-	}
-
-	netstr := fmt.Sprintf("%s/%d", cfg.IPAddress, cfg.IPMask)
-	ip, ipNet, err := net.ParseCIDR(netstr)
-	if err != nil {
-		return nil, fmt.Errorf("not a valid IP address: %s, err: %v", netstr, err)
-	}
-
 	if len(cfg.Disk) < 1 {
 		return nil, fmt.Errorf("No disk provided")
 	}
 
-	gw := net.ParseIP(cfg.GateWay)
+	var (
+		ipAddress net.IP
+		ipMask    net.IPMask
+		gateway   net.IP
+		tapName   string
+		mac       string
+		err       error
+	)
+
+	// If network details are specified, use them, if not do usual network plumbing
+	if len(cfg.IPAddress) > 0 && len(cfg.Gateway) > 0 && len(tapName) > 0 {
```

Comment:
What if just one or two of the 3 are specified, like everything defined except gateway. If that's invalid config, an error should be returned i guess, right?

Reply:
If several of them are not specified, we have some defaults put in in the last else clause statement (this is the original runnc-cont behavior). I moved the default logic back down to the bottom else clause.

Reply:
it probably could be neater, but i didn’t tidy it up because im removing that entire bottom section in the next PR, and just doing what you said that we just return an error if not all the fields are valid.